### PR TITLE
GD-200: Handle null values on SceneRunner `get_property`

### PR DESCRIPTION
**Pull request by clemens-tolboom**
To fix this we need to 'know' the scene properties.

### CODE REVIEW

File: addons/gdUnit4/src/core/GdUnitSceneRunnerImpl.gd [hunk 1]
```diff
@@ -240,11 +240,21 @@ func maximize_view() -> GdUnitSceneRunner:
...
-	var property = _current_scene.get(name)
-	if property != null:
-		return property
+	if __properties.size() == 0:
+		__build_properties()
+
+	if __properties.has(name):
+		return _current_scene.get(name)
+
 	return  "The property '%s' not exist checked loaded scene." % name
```
Comment (MikeSchulze) [inline addons/gdUnit4/src/core/GdUnitSceneRunnerImpl.gd:258]:
```suggestion
func get_property(name :String) -> Variant:
	if not _property_exists(name):
		return "The property '%s' not exist checked loaded scene." % name
	return _current_scene.get(name)
```

File: addons/gdUnit4/src/core/GdUnitSceneRunnerImpl.gd [hunk 2]
```diff
@@ -240,11 +240,21 @@ func maximize_view() -> GdUnitSceneRunner:
 	DisplayServer.window_move_to_foreground()
 	return self
 
+# Property cache
+var __properties:Dictionary
+
+# Helper function to fill __properties
+func __build_properties():
+	if __properties.size() == 0:
+		_current_scene.get_property_list().map(func(p:Dictionary) : __properties[p.name] = p)
```
Comment (MikeSchulze) [inline addons/gdUnit4/src/core/GdUnitSceneRunnerImpl.gd:249]:
```suggestion
func _property_exists(name :String) -> bool:
	return _current_scene.get_property_list().any(func(properties :Dictionary) : return properties["name"] == name)
```

**Review by MikeSchulze — CHANGES_REQUESTED** [review]
there is no need to cache properties use instead a filter ;)
Please add also unit tests to cover this.

**Comment by clemens-tolboom** [comment]
Superseded by #202 ... closing.

**Pull request closed without merging**